js/console_test.ts

```diff
@@ -310,3 +318,272 @@ test(function consoleDetachedLog() {
   consoleGroupEnd();
   consoleClear();
 });
+
+class StringBuffer {
+  chunks: string[] = [];
+  add(x: string) {
+    this.chunks.push(x);
+  }
+  toString() {
+    return this.chunks.join("");
+  }
+}
+
+type ConsoleExamineFunc = (
+  csl: any, // tslint:disable-line:no-any
+  out: StringBuffer,
+  err?: StringBuffer,
+  both?: StringBuffer
+) => void;
+
+function mockConsole(f: ConsoleExamineFunc) {
+  const out = new StringBuffer();
+  const err = new StringBuffer();
+  const both = new StringBuffer();
+  const csl = new Console(
+    (x: string, isErr: boolean, printsNewLine: boolean) => {
+      const content = x + (printsNewLine ? "\n" : "");
+      const buf = isErr ? err : out;
+      buf.add(content);
```

Comment:
Hi! Sorry for the seemingly "drive-by" review question, but here goes: what is `buf` used for?

Reply:
For faking either stdout or stderr. I want to be able to view each log attempts in order through the chunks